### ISSUE TITLE
Parser errors for some var functions

### ISSUE THREAD

**Issue by winmain** [HaxeFoundation/intellij-haxe#521]
![image](https://cloud.githubusercontent.com/assets/167248/20477023/be12cc2a-b00e-11e6-9f2b-fe9728d772e8.png)

`fn1` brokes when var declares a function with `Void` arguments. Any other argument type works fine.

`fn2` brokes everything below when function declared without curly braces. Workaround here is to use function with body declared in curly braces, for example: `function() {return 5;}`

```haxe
class Test {

  var fn1: Void -> Void = function(): Void {}; // Incompatible type error for Void argument functions

  var fn2: Void -> Int = function() return 5;

  public static function main(): Void { // "public" unexpected after var with short function
  }
}
```

Plugin version: `0.10.1-RC2`
Idea: `2016.2`

**Comment by mayakwd** [comment]
```
var fn2: Void -> Int = function() return 5;

public static function main(): Void { // "public" unexpected after var with short function
```

This block is parsing incorrectly because fn2 is awaiting for closing semicolon, existing semicolon is belongs to returnStatement (from functionCommonBody)

**Comment by EricBishton** [comment]
The fn1 error was fixed by @mayakwd's submission.  Only `fn2` left...

**Comment by EricBishton** [comment]
fn2 was fixed at some point during 0.11.2 release cycle.
